### PR TITLE
[Dashboard] ERC20 burn tab

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
The focus of this PR is to update token transfer and burn functionalities in the dashboard UI.

### Detailed summary
- Updated icons and components for token transfer and burn actions
- Replaced `useTxNotifications` with manual error handling and toast notifications
- Improved UI layout and added form validation for token burn action

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4661?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4661** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4661?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4643** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4643?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx [hunk 1]
```diff
@@ -24,27 +48,112 @@ export const TokenBurnButton: React.FC<TokenBurnButtonProps> = ({
...
   const hasBalance = tokenBalanceQuery.data && tokenBalanceQuery.data > 0n;
+  const [open, setOpen] = useState(false);
+  const sendConfirmation = useSendAndConfirmTransaction();
+  const trackEvent = useTrack();
+  const form = useForm({ defaultValues: { amount: "0" } });
+  const { onSuccess, onError } = useTxNotifications(
+    "Tokens burned successfully",
+    "Failed to burn tokens",
+    contract,
+  );
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx:59]: prefer removing this whenever u come across it and use `toast.promise()` instead

File: apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx [hunk 2]
```diff
@@ -1,20 +1,44 @@
...
+import { useState } from "react";
+import { useForm } from "react-hook-form";
 import { FaBurn } from "react-icons/fa";
-import type { ThirdwebContract } from "thirdweb";
+import { type ThirdwebContract, toUnits } from "thirdweb";
 import { balanceOf } from "thirdweb/extensions/erc20";
-import { useActiveAccount, useReadContract } from "thirdweb/react";
-import { Button, Drawer } from "tw-components";
-import { TokenBurnForm } from "./burn-form";
+import * as ERC20Ext from "thirdweb/extensions/erc20";
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx:18]: double import, can just use `ERC20Ext.balanceOf`

File: apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx [hunk 3]
```diff
@@ -1,20 +1,44 @@
...
+  SheetTitle,
+  SheetTrigger,
+} from "@/components/ui/sheet";
+import { FormControl, Icon, Input, Stack } from "@chakra-ui/react";
+import { TransactionButton } from "components/buttons/TransactionButton";
+import { useTrack } from "hooks/analytics/useTrack";
+import { useTxNotifications } from "hooks/useTxNotifications";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
 import { FaBurn } from "react-icons/fa";
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx:15]: whenever u see a `react-icons` import feel free to replace it with an appropriate icon from `lucide-react` please :)

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx (hunk 1 above)
Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx:58]: @jnsdls do we also do `useInvalidateContractQuery` like how it's setup inside useTxNotification at the moment?

Comment (jnsdls) [reply apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx:58]: no we should no longer need it, v5 sdk handles it for us

File: apps/dashboard/src/contract-ui/tabs/tokens/components/transfer-button.tsx [hunk 4]
```diff
@@ -141,7 +135,8 @@ export const TokenTransferButton: React.FC<TokenTransferButtonProps> = ({
                     label: "error",
                     error,
                   });
-                  onError(error);
+                  toast.error("Failed to transfer tokens");
+                  console.error(error);
                 },
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/tokens/components/transfer-button.tsx:140]: <i class='graphite__hidden'>[Re: lines +120 to +140]<br/><br/></i>

use `toast.promise` instead on `sendConfirmation.mutateAsync()`, this way you get a loading state and a success and error state that is consistent<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4661?utm_source=unchanged-line-comment">Graphite</a>.</span>

File: apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx [hunk 5]
```diff
@@ -1,50 +1,158 @@
...
+                    return {
+                      amount: toUnits(
+                        data.amount,
+                        await ERC20Ext.decimals({ contract }),
+                      ),
+                    };
+                  },
+                });
+
+                sendConfirmation.mutate(tx, {
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/tokens/components/burn-button.tsx:128]: use mutateAsync and pass the resulting promise into `toast.promise()`

**Review by jnsdls — CHANGES_REQUESTED** [review]

**Review by jnsdls — APPROVED** [review]

**Pull request merged**